core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/RebindIteration.java

```diff
@@ -467,6 +469,15 @@ protected void instantiateMementos() throws IOException {
         memento = persistenceStoreAccess.loadMemento(mementoRawData, rebindContext.lookup(), exceptionHandler);
     }
 
+    protected void initPlaneId() {
+        String persistedPlaneId = mementoRawData.getPlaneId();
+        if (persistedPlaneId == null) {
+            ((LocalManagementContext)managementContext).generateManagementPlaneId();
```

Comment:
On balance, I agree with this code. But it feels like we're relying on null too much. We don't distinguish between persisted state that has no entities etc, and an unpopulated persisted state directory - that makes it hard to write code that asserts/warns it's in the state we expect.